staging/test-resources/repl-staging/i6007

```diff
@@ -1,7 +1,7 @@
 scala> import scala.quoted._
-scala> import scala.quoted.staging._
-scala> implicit def toolbox: Toolbox = Toolbox.make(getClass.getClassLoader)
-def toolbox: quoted.staging.Toolbox
+scala> import quoted.staging.{Compiler => StagingCompiler, _}
+scala> implicit def compiler: StagingCompiler = StagingCompiler.make(getClass.getClassLoader)
```

Comment:
I had to rename it here due to some strange interaction with `java.lang.Compiler` import

```
scala> import quoted._
scala> import quoted.staging._
scala> implicit def compiler: Compiler = Compiler.make(getClass.getClassLoader)
1 | implicit def compiler: Compiler = Compiler.make(getClass.getClassLoader)
  |                        ^^^^^^^^
  |                       Reference to Compiler is ambiguous,
  |                       it is both imported by import quoted.staging._
  |                       and imported subsequently by import java.lang.{...}
```

Reply:
`StagingCompiler` sounds like an even better name than `Compiler`.

Reply:
This is a more general issue. I opened #11146.

Reply:
If staging compiler sound better, maybe we could rewrite the docs to use
```scala
import scala.quoted._

given staging.Compiler = staging.Compiler.make(...)

staging.run {
  ...
}
```

That does not affect the API changes and could be done later.

Reply:
I think it will help in avoiding misuse of the staging API --- the users are forced to think what's staging and its difference from macros.